src/main/java/com/greghaskins/spectrum/internal/Spec.java

```diff
@@ -1,18 +1,23 @@
 package com.greghaskins.spectrum.internal;
 
-import com.greghaskins.spectrum.internal.blocks.NotifyingBlock;
+
+import com.greghaskins.spectrum.Block;
+import com.greghaskins.spectrum.internal.hooks.Hook;
+import com.greghaskins.spectrum.internal.hooks.HookContext;
+import com.greghaskins.spectrum.internal.hooks.Hooks;
 
 import org.junit.runner.Description;
 import org.junit.runner.notification.Failure;
 
-final class Spec implements Child {
+final class Spec implements LeafChild {
 
-  private final NotifyingBlock block;
+  private final Block block;
```

Comment:
The block is kept here in its raw form, as the leaf hooks will wrap it with reporting.